flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/ClusterHealthEvaluator.java

```diff
@@ -75,72 +77,139 @@ public void evaluate(
             var lastValidClusterHealthInfo = getLastValidClusterHealthInfo(clusterInfo);
             if (lastValidClusterHealthInfo == null) {
                 LOG.debug("No last valid health info, skipping health check");
```

Comment:
I think we should clear all health info (restarts, checkpoint progress) during an upgrade/suspend, so we guarantee that it is reinitialized after the jm starts after the upgrade.

Reply:
Added.